### PR TITLE
`Communication`: Display improved edited hints

### PR DESCRIPTION
**Pull request by TimOrtel**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

Currently, the server simply appends (last edited by xxx on date yyy) to the content of edited messages. We want to improve this.

### Description

1. Added new column updated_date to posting to store if the posting has ever been updated and when so.
2. The client displays (edited) on posts that have been edited.

### Steps for Testing
1. Go to conversations and send a new message
2. Edit the message previously sent
3. You should see (edited) on the post

#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

image

### CODE REVIEW

File: src/main/webapp/app/shared/metis/posting-content/posting-content.component.html [hunk 1]
```diff
@@ -9,10 +9,14 @@
         </span>
         <div *ngIf="showContent">
             <jhi-posting-content-part *ngFor="let postingContentPart of postingContentParts" [postingContentPart]="postingContentPart"> </jhi-posting-content-part>
+
+            <span class="edited-text" *ngIf="isEdited">(edited)</span>
```
Comment (sven0311) [inline src/main/webapp/app/shared/metis/posting-content/posting-content.component.html:13]: shouldn't you use the translated text here as well?

**Comment by sven0311** [comment]
I think a liquibase migration is missing

**Review by sven0311 — COMMENTED** [review]
Tested on TS1. Looks really good now

File: src/main/webapp/app/shared/metis/posting-content/posting-content.component.html (hunk 1 above)
Comment (TimOrtel) [reply src/main/webapp/app/shared/metis/posting-content/posting-content.component.html]: Good catch! I updated it.

**Comment by sven0311** [comment]
> I think a liquibase migration is missing

Nevermind, I just didn't see it 😅

**Review by sven0311 — DISMISSED** [review]
Looks good now! Ready to roll

**Review by Strohgelaender — CHANGES_REQUESTED** [review]
Pinning a post shows it as edited: 

https://github.com/ls1intum/Artemis/assets/26540346/01e416a2-486b-48a6-826a-02f662d2a891

Same happens when accepting a AnswerPost as resolving.

**Review by Strohgelaender — DISMISSED** [review]
looks good now

**Review by terlan98 — APPROVED** [review]
Code LGTM

**Review by laadvo — DISMISSED** [review]
Tested on ts1, the edited hint is displayed as expected

**Review by sven0311 — APPROVED** [review]
reapprove

**Review by dearjasmina — DISMISSED** [review]
tested on ts1, works as expected

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
When you load from or send the date to the server, you should use convertDateFromClient/convertDateFromServer in the client. Otherwise methods like isBefore() don't work since the dates would only be strings and not actual date objects

**Comment by undernagruzez** [comment]
Tested on ts2. Can confirm that pinning a post causes an 'edited' label to appear. When reloading the page or repinning, the state is shown correctly.

**Comment by TimOrtel** [comment]
> 
Thank you for your comment @undernagruzez. Sadly, I cannot reproduce the behavior you described from your description. Could you please append a detailed video where this happens?

**Comment by undernagruzez** [comment]
> Details

Retested on ts1, was not able to reproduce that behavior.

**Review by JohannesStoehr — DISMISSED** [review]
Thanks for addressing my comment. Code looks good to me

**Review by eceeeren — DISMISSED** [review]
Tested on TS2, works well

**Review by asliayk — APPROVED** [review]
tested on ts2, works as expected

**Review by Strohgelaender — DISMISSED** [review]
lgtm

**Review by krusche — APPROVED** [review]
Changes look good 👍 The last commit only merged a conflict in `master.xml`
When the checks pass, we can merge the PR

**Pull request merged**